integrations/thegraph.md

```diff
@@ -0,0 +1,265 @@
+      apiVersion: 0.0.4
+      language: wasm/assemblyscript
+      file: ./src/mapping.ts
+      entities:
+        - Player
+        - Round
+        - Ticket
+        - Winner
+      abis:
+        - name: MoonLotto
+          file: ./artifacts/contracts/MoonLotto.sol/MoonLotto.json
+      eventHandlers:
+        - event: PlayerJoined(uint256,address,uint256,bool,uint256)
+          handler: handlePlayerJoined
+        - event: LotteryResult(uint256,address,uint256,bool,uint256,uint256)
+          handler: handleLotteryResult
+```
+
+### Mappings
+
+Mappings files are what transform the blockchain data into entities defined in the schema file. Each event handler inside the YAML file needs to have a subsequent function inside the mapping file.
+
+The mapping file used for the Lottery example can be found in [this link](https://github.com/PureStake/moonlotto-subgraph/blob/main/src/mapping.ts).
+
+In general, the strategy of each handler function is to load the event data, check if an entry already exists, arrange the data as desired, and save the entry. For example, the handler function for the `PlayerJoined` event is as follows:
+
+```
+export function handlePlayerJoined(event: PlayerJoined): void {
+  // ID for the round:
+  // round number
+  let roundId = event.params.round.toString();
+  // try to load Round from a previous player
+  let round = Round.load(roundId);
+  // if round doesn't exists, it's the first player in the round -> create round
+  if (round == null) {
+    round = new Round(roundId);
+    round.timestampInit = event.block.timestamp;
+  }
+  round.index = event.params.round;
+  round.prize = event.params.prizeAmount;
+
+  round.save();
+
+  // ID for the player:
+  // issuer address
+  let playerId = event.params.player.toHex();
+  // try to load Player from previous rounds
+  let player = Player.load(playerId);
+  // if player doesn't exists, create it
+  if (player == null) {
+    player = new Player(playerId);
+  }
+  player.address = event.params.player;
+
+  player.save();
+
+  // ID for the ticket (round - player_address - ticke_index_round):
+  // "round_number" + "-" + "player_address" + "-" + "ticket_index_per_round"
+  let nextTicketIndex = event.params.ticketIndex.toString();
+  let ticketId = roundId + "-" + playerId + "-" + nextTicketIndex;
+
+  let ticket = new Ticket(ticketId);
+  ticket.round = roundId;
+  ticket.player = playerId;
+  ticket.isGifted = event.params.isGifted;
+  ticket.isWinner = false;
+
+  ticket.save();  
+}
+```
+
+## Deploying a Subgraph
+
+If you are going to use The Graph API (hosted service), you need to (all steps can be found in [this link](https://thegraph.com/docs/deploy-a-subgraph)):
+
+ - Create a [Graph Explorer](https://thegraph.com/explorer/) account, you will need a Github account
+ - Go to your dashboard and write down the access token
+ - If using The Graph API, you need to create your Subgraph via the "Add Subgraph" button in the Graph Explorer site
```

Comment:
You refer twice here to use The Graph API when it's already its section.
```suggestion
 - Create your Subgraph via the "Add Subgraph" button in the Graph Explorer site
```

Reply:
good catch